receiver/snmpreceiver/config.go

```diff
@@ -14,17 +14,578 @@
+	// Valid options: “no_auth_no_priv”, “auth_no_priv”, “auth_priv”
+	// Default: "no_auth_no_priv"
+	SecurityLevel string `mapstructure:"security_level"`
+
+	// AuthType is the type of authentication protocol to use for this SNMP connection.
+	// Only valid for version “v3” and if “no_auth_no_priv” is not selected for SecurityLevel
+	// Valid options: “md5”, “sha”, “sha224”, “sha256”, “sha384”, “sha512”
+	// Default: "md5"
+	AuthType string `mapstructure:"auth_type"`
+
+	// AuthPassword is the authentication password used for this SNMP connection.
+	// Only valid for version "v3" and if "no_auth_no_priv" is not selected for SecurityLevel
+	AuthPassword string `mapstructure:"auth_password"`
+
+	// PrivacyType is the type of privacy protocol to use for this SNMP connection.
+	// Only valid for version “v3” and if "auth_priv" is selected for SecurityLevel
+	// Valid options: “des”, “aes”, “aes192”, “aes256”, “aes192c”, “aes256c”
+	// Default: "des"
+	PrivacyType string `mapstructure:"privacy_type"`
+
+	// PrivacyPassword is the authentication password used for this SNMP connection.
+	// Only valid for version “v3” and if "auth_priv" is selected for SecurityLevel
+	PrivacyPassword string `mapstructure:"privacy_password"`
+
+	// ResourceAttributes defines what resource attributes will be used for this receiver
+	ResourceAttributes map[string]*ResourceAttributeConfig `mapstructure:"resource_attributes"`
+
+	// Attributes defines what attributes will be used on metrics for this receiver
+	Attributes map[string]*AttributeConfig `mapstructure:"attributes"`
+
+	// Metrics defines what SNMP metrics will be collected for this receiver
+	Metrics map[string]*MetricConfig `mapstructure:"metrics"`
+}
+
+// ResourceAttributeConfig contains config info about all of the resource attributes that will be used by this receiver.
+type ResourceAttributeConfig struct {
+	// Description is optional and describes what the resource attribute represents
+	Description string `mapstructure:"description"`
+	// OID is required only if IndexedValuePrefix is not defined.
+	// This is the column OID which will provide indexed values to be used for this resource attribute (alongside a metric with ColumnOIDs)
+	// in order to create multiple "resources" for each indexed metric value
+	OID string `mapstructure:"oid"`
+	// IndexedValuePrefix is required only if OID is not defined.
+	// This is used alongside metrics with ColumnOIDs to assign resource attribute values using this prefix + the OID index of the metric value.
+	// This will result in multiple "resources" being created for each indexed metric value
+	IndexedValuePrefix string `mapstructure:"indexed_value_prefix"` // required and valid if no oid field
+}
+
+// AttributeConfig contains config info about all of the metric attributes that will be used by this receiver.
+type AttributeConfig struct {
+	// Value is optional, and will allow for a different attribute key value other than the attribute name
```

Comment:
```suggestion
	// Value is optional, and will allow for a different attribute value other than the attribute name
```

We need to be clear about whether this is "name" (key) or "value". Should this be "name"?